numpy/core/src/common/ufunc_override.c

```diff
@@ -97,17 +99,21 @@ PyUFuncOverride_GetOutObjects(PyObject *kwds, PyObject **out_kwd_obj, PyObject *
          * The C-API recommends calling PySequence_Fast before any of the other
          * PySequence_Fast* functions. This is required for PyPy
          */
-        PyObject *seq = PySequence_Fast(*out_kwd_obj, "Could not convert object to sequence");
+        PyObject *seq;
         int ret;
+        seq = PySequence_Fast(*out_kwd_obj,
+                              "Could not convert object to sequence");
         if (seq == NULL) {
+            *out_kwd_obj = NULL;
             return -1;
         }
         *out_objs = PySequence_Fast_ITEMS(seq);
         ret = PySequence_Fast_GET_SIZE(seq);
-        Py_SETREF(*out_kwd_obj, seq);
+        *out_kwd_obj = seq;
         return ret;
```

Comment:
nit: you can now reorder these to eliminate the `ret` variable.

Reply:
ok